torch/csrc/jit/codegen/cuda/python_frontend/examples/python_example_fp16.py

```diff
@@ -0,0 +1,33 @@
+import torch
+
+from torch._C._nvfuser import FusionDefinition,Fusion,DataType
+from torch._C._nvfuser.state_creation import make_tensor,make_constant
+from torch._C._nvfuser.ops import *
+
+### Construct Fusion 
+fusion = Fusion()
+
+with FusionDefinition(fusion) as fd :
+    t0 = make_tensor(2, DataType.Half)
+    t1 = make_tensor(2, DataType.Half)
+
+    fd.add_input(t0)
+    fd.add_input(t1)
+
+    t2 = add(t0, t1)
+    t3 = cast(DataType.Half, t2)
```

Comment:
I like the part here that we expose the cast explicitly.

I think one thing we need to communicate with upstream is type propagation requirement. Currently there's no ground truth where we can infer the `consistent` output scalar type.